File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        ans="".join(i for i in s if i.isalnum())
        ans=ans.lower()
        st=0
        end=len(ans)-1
        while(st<=end):
            if ans[st]!=ans[end]:
                return False
            st+=1
            end-=1
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str = new StringBuilder();
        for(int i=0;i<s.length();i++){
            char curr = s.charAt(i);
            if(Character.isLetterOrDigit(curr)){
                str.append(curr);
            }
        }
        String ans = str.toString().toLowerCase();
        int l=0;
        int r = ans.length()-1;
        while(l<=r){
            if(ans.charAt(l) != ans.charAt(r)){
                return false;
            }
            l+=1;
            r-=1;
        }
        return true;

    }
}